src/script/event/preprocessor/QuotedMessageMiddleware.ts

```diff
@@ -104,37 +103,27 @@ export class QuotedMessageMiddleware {
 
     const messageId = quote.quotedMessageId;
 
-    const quotedMessage = await this.eventService.loadEvent(event.conversation, messageId);
+    let quotedMessage = await this.eventService.loadEvent(event.conversation, messageId);
     if (!quotedMessage) {
-      this.logger.warn(`Quoted message with ID "${messageId}" not found.`);
-      const quoteData = {
-        error: {
-          type: QuoteEntity.ERROR.MESSAGE_NOT_FOUND,
-        },
-      };
-
-      const decoratedData = {...event.data, quote: quoteData};
-      return {...event, data: decoratedData};
+      const replacedMessage = await this.eventService.loadReplacingEvent(event.conversation, messageId);
+      if (!replacedMessage) {
+        this.logger.warn(`Quoted message with ID "${messageId}" not found.`);
+        const quoteData = {
+          error: {
+            type: QuoteEntity.ERROR.MESSAGE_NOT_FOUND,
+          },
+        };
+
+        const decoratedData = {...event.data, quote: quoteData};
+        return {...event, data: decoratedData};
+      }
+      quotedMessage = replacedMessage;
     }
 
-    const quotedMessageHash = new Uint8Array(quote.quotedMessageSha256).buffer;
-
-    const isValid = await MessageHasher.validateHash(quotedMessage, quotedMessageHash);
```

Comment:
Can you check if the messageHasher could not be removed from the codebase at all?

Reply:
i dont believe that is the case, this was something from the original testing, and already approved by security. in some cases the hash will simply not match. there's really no way around it if we want to solve the original 'message not found bug'

Reply:
if you send a hash of a different message it must fail, even if we see the replacing id being updated.

Reply:
Oh, it's still used for sending quotes, right?

Reply:
yep